idl/testing/run_tests.py

```diff
@@ -0,0 +1,1218 @@
+socket=          31
+socket=          32
+socket=          33
+socket=          34
+socket=          35
+socket=          36
+socket=          37
+socket=          38
+socket=          39
+socket=          40
+socket=          41
+socket=          42
+socket=          43
+socket=          44
+socket=          45
+socket=          46
+socket=          47
+socket=          48
+socket=          49
+socket=          50
+socket=          51
+socket=          52
+socket=          53
+socket=          54
+socket=          55
+socket=          56
+socket=          57
+socket=          58
+socket=          59
+socket=          60
+socket=          61
+socket=          62
+socket=          63
+socket=          64
+socket=          65
+socket=          66
+socket=          67
+socket=          68
+socket=          69
+socket=          70
+socket=          71
+socket=          72
+socket=          73
+socket=          74
+socket=          75
+socket=          76
+socket=          77
+socket=          78
+socket=          79
+socket=          80
+socket=          81
+socket=          82
+socket=          83
+socket=          84
+socket=          85
+socket=          86
+socket=          87
+socket=          88
+socket=          89
+socket=          90
+socket=          91
+socket=          92
+socket=          93
+socket=          94
+socket=          95
+socket=          96
+socket=          97
+socket=          98
+socket=          99
+SUCCESS
+
+''')
+
+
+# Issue #2639: mdsvalue works without a socket
+idl_test(f'''
+         
+testid = 'IDL-2639-no-socket'
+PASS = 1
+FAIL = 0
+DATA = '55'
+mdsip_server = '{server}'
+test_status = PASS
+
+result = mdsvalue(DATA)
+print, 'result=', result
+if (result ne fix(DATA)) then test_status = FAIL
+
+if (test_status eq PASS) then begin
+    print, 'SUCCESS'
+end else begin
+    print, 'FAILURE'
+endelse
+
+''',
+'''
+
+% Compiled module: MDSVALUE.
+% Compiled module: MDSCHECKARG.
+% Compiled module: MDSISCLIENT.
+% Compiled module: MDS_KEYWORD_SET.
+% Compiled module: MDSIDLIMAGE.
+result=          55
+SUCCESS
+
+''')
+
+
+# Issue #2639: mdsvalue and interaction with killed socket.
+# idl_test(f'''
+         
+# testid = 'IDL-2639-kill-last-socket'
+# PASS = 1
+# FAIL = 0
+# DATA = '55'
+# mdsip_server = '{server}'
+# test_status = PASS
+
+# mdsconnect, mdsip_server
+# mdsconnect, mdsip_server
+
+# result = mdsvalue(DATA)
+# print, 'result=', result
+# if (result ne fix(DATA)) then test_status = FAIL
+
+# mdsdisconnect, socket=1
+
+# result = mdsvalue(DATA)
+# print, 'result=', result
+# if (result ne fix(DATA)) then test_status = FAIL
+
+# if (test_status eq PASS) then begin
+#     print, 'SUCCESS'
+# end else begin
+#     print, 'FAILURE'
+# endelse
+
+# ''',
+# '''
+
+# % Compiled module: MDSCONNECT.
+# % Compiled module: MDS_KEYWORD_SET.
+# % Compiled module: MDSDISCONNECT.
+# % Compiled module: MDSVALUE.
+# % Compiled module: MDSCHECKARG.
+# % Compiled module: MDSISCLIENT.
+# result=          55
```

Comment:
Why was this test commented out?

Reply:
See explanation given two posts below (25-Oct-2023 11:58 a.m. PDT).